deploy/chart/templates/plugin.yaml

```diff
@@ -1,17 +1,20 @@
-{{- if .Values.plugin.enabled -}}
+{{- range .Values.nodepools -}}
+{{- if $.Values.plugin.enabled }}
+{{- $copy := deepCopy .csi }}
+{{- $csi := mergeOverwrite $.Values.csi $copy }}
 kind: DaemonSet
 apiVersion: apps/v1
 metadata:
-  name: csi-plugin
-  namespace: {{ .Release.Namespace }}
+  name: {{ .name }}
```

Comment:
Can we just use the key in .Values.nodepools as the name? e.g. 
```helm
{{ if eq $key "default" }}
name: csi-plugin
{{ else }}
name: csi-pluign-{{ $key }}
{{ end }}
```
We can avoid possible duplicated name in this way.

Reply:
I personally think it's more readable to explicitly clarify names of each node pool. It makes no real difference compared to using $key as we do now.

Reply:
OK. But this way we lost a level of abstraction. e.g. `kubectl get pods -l nodepool=edge` seems more reasonable than `-l nodepool=csi-plugin-edge`